tlatools/org.lamport.tlatools/src/tlc2/debug/TLCSourceBreakpoint.java

```diff
@@ -27,20 +27,26 @@
 
 import org.eclipse.lsp4j.debug.SourceBreakpoint;
 
+import tla2sany.semantic.ModuleNode;
+import tla2sany.semantic.OpDefNode;
 import tla2sany.st.Location;
+import tlc2.tool.impl.SpecProcessor;
 
 public class TLCSourceBreakpoint extends SourceBreakpoint {
 
 	private final int hits;
 	private final Location location;
+	// Use Getter
+	public final OpDefNode condition;
 	
-	public TLCSourceBreakpoint(final String module, final SourceBreakpoint s) {
+	public TLCSourceBreakpoint(final SpecProcessor processor, final String module, final SourceBreakpoint s,
+			final ModuleNode semanticRoot) {
 		setColumn(s.getColumn());
 		setLine(s.getLine());
 		// Create a location that's not a point.
 		final int column = getColumn() != null ? getColumn() : 1;
 		//TODO: If the location spans lines, getLine() + 1 should be the endline (second line parameter of Location).
-		location = new Location(module, getLine() + 1, column, getLine(), column + 1);
+		location = new Location(module, getLine(), column, getLine(), column + 1);
```

Comment:
This change is likely the cause of the test failures. I believe there was a mismatch between the SANY and the DAP locations.

Reply:
I recall I ran into trouble getting a match using the `SemanticNode.pathTo` method with the `+ 1` added; will investigate further.

Reply:
Related, breakpoint validation/verification (hollow breakpoint vs. Red breakpoint) no longer seems to works.  Bisecting should be the fastest to find the offending commit

Reply:
Like bisect within these changes or in the repo in general?

Reply:
It seems like the `+ 1` works with my changes to how `pathTo` works (accept if node is a leaf node and the location is contained within it).